src/Resources/Chat.php

```diff
@@ -49,6 +50,26 @@ public function createStreamed(array $parameters): StreamResponse
 
         $response = $this->transporter->requestStream($payload);
 
-        return new StreamResponse(CreateStreamedResponse::class, $response);
+        $streamResponse = new StreamResponse(CreateStreamedResponse::class, $response);
+
+        $this->assertValidResponse($streamResponse, $parameters['model']);
+
+        return $streamResponse;
+    }
+
+    /**
+     * Asserts that the given StreamResponse is valid by checking its status code.
+     * Throws a NotFoundException if the status code is 404.
+     *
+     * @param StreamResponse $streamResponse The StreamResponse to be validated
+     * @param string         $model          The model name associated with the StreamResponse
+     *
+     * @throws NotFoundException If the StreamResponse's status code is 404
+     */
+    private function assertValidResponse(StreamResponse $streamResponse, string $model): void
+    {
+        if (404 === $streamResponse->getStatusCode()) {
```

Comment:
To check only against the status code is maybe a bit unsecure. Probably the API will return a 404 in more different states, but they did not document their usage of a 404: 

I think we should check against the response content as the exact error is reported there:

```php
try {
    $response = ($this->streamHandler)($request);
} catch (ClientExceptionInterface $clientException) {
    dd($clientException->getResponse()->getBody()->getContents()); // output see below
    throw new TransporterException($clientException);
}
```

```json
{
    "error": {
        "message": "The model `gpt-42` does not exist",
        "type": "invalid_request_error",
        "param": null,
        "code": null
    }
}
```

@nunomaduro Maybe we should improve the error handling in general to use the error message returned from the API if available.

Reply:
Currently the error is not coming from the HttpTransporter. I would handle the change iteratively, and once you have reported the bug to OpenAI we can go back to the drawing board. Then you could use the message from the origin API.

Reply:
@IshtarStar Sorry, but I don't get it.

What is the error on OpenAI side? They do return a status code 404 which is clearly not a successful response.

Could you please explain how the OpenAI response should look like in your opinion?

Reply:
We can close this PR. See new:
https://github.com/openai-php/client/pull/90